### PR TITLE
Fix inconsistent shadows

### LINKED ISSUE #824: Component shadows do not always applies in 0.63.37
### Environment
react-native -v: 0.63.4
npm ls react-native-macos: 0.63.37
node -v: 15.14.0
npm -v: 7.7.8
yarn --version: -
xcodebuild -version: 12.5.1

### Issue
Component shadows will not be visible half of the time starting the app

### Steps to Reproduce
1. Init a new project using the docs
2. Add a view component with shadow property
```
const App = () => {
  return (
    <>
      <View style={s.view}>
        <View style={s.shadow} />
      </View>
    </>
  );
};

const s = StyleSheet.create({
  view: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  viewWithShadow: {
    width: 200,
    height: 100,
    backgroundColor: 'blue',
    shadowRadius: 8,
    shadowColor: '#000000',
    shadowOpacity: 0.6,
    shadowOffset: {
      height: 6,
      width: 0,
    },
  },
});
```
3. Either build the app using development mode or production mode
4. Start the app multiple times and watch the shadow not working about half of the time

### Expected Behavior
Shadows always being applied

### Actual Behavior
Shadows only being applied 50% of the time

### PR DESCRIPTION
**Pull request by lyzhan7**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

Pull request #1227 mostly fixed the consistency issue but had a problem with the shadow opacity. There isn't a shadowOpacity property on NSShadow, the shadow opacity from the layer needed to be passed into the alpha channel of the NSShadow shadowColor property.

Fixes #824 

## Test Plan

This fix was tested with the fluentui-react-native tester app.

Before:

https://user-images.githubusercontent.com/78454019/184040031-20bab19d-8ed6-42ff-98aa-c04977b8d786.mov

After:

https://user-images.githubusercontent.com/78454019/184040044-730c62db-f871-4007-80b8-0cb7945bbea0.mov

### CODE REVIEW

**Comment by lyzhan7** [comment]
Note: We tried an additional approach to fix this, since the root issue behind this bug is that modifying certain layer props (including shadow related ones) on a layer owned by an NSView is undefined behavior, and we are still doing this in RCTViewManager with RCT_REMAP_VIEW_PROPERTY

```
RCT_REMAP_VIEW_PROPERTY(shadowColor, layer.shadowColor, CGColor)
RCT_REMAP_VIEW_PROPERTY(shadowOffset, layer.shadowOffset, CGSize)
RCT_REMAP_VIEW_PROPERTY(shadowOpacity, layer.shadowOpacity, float)
RCT_REMAP_VIEW_PROPERTY(shadowRadius, layer.shadowRadius, CGFloat)
```

In the second approach, rather than modifying RCTView.m, we changed RCTViewManager.m to use RCT_CUSTOM_VIEW_PROPERTY instead of RCT_REMAP_VIEW_PROPERTY, and set the view.shadow there.

MicrosoftTeams-image

We decided to keep the original approach since the result is the same, it's simpler and there will be less diffs between React Native macOS and React Native Core. If there are further issues with shadow rendering consistency this could be something we revisit.

File: React/Views/RCTView.m [hunk 1]
```diff
@@ -716,6 +716,21 @@ - (NSString *)description
...
 
 #if TARGET_OS_OSX // [TODO(macOS GH#774)
+- (NSShadow*)shadow {
+  CALayer *layer = self.layer;
+
+  // Workaround AppKit issue with directly manipulating the view layer's shadow.
+  if (layer.shadowColor != nil && layer.shadowOpacity > 0) {
+    NSShadow *shadow = [NSShadow new];
+
+    shadow.shadowColor = [[NSColor colorWithCGColor:layer.shadowColor] colorWithAlphaComponent:layer.shadowOpacity];
```
Comment (Saadnajmi) [inline React/Views/RCTView.m:726]: @lyahdav Turns out we just needed the extra `withAlphaComponent` bit here :)

File: React/Views/RCTView.m [hunk 2]
```diff
@@ -716,6 +716,21 @@ - (NSString *)description
 #endif // TODO(macOS GH#774)
 
 #if TARGET_OS_OSX // [TODO(macOS GH#774)
+- (NSShadow*)shadow {
+  CALayer *layer = self.layer;
+
+  // Workaround AppKit issue with directly manipulating the view layer's shadow.
```
Comment (Saadnajmi) [inline React/Views/RCTView.m:722]: Nit: I think the comment is better of closer to the #IF TARGET_OS_OSX tag, aka put it above line 719

File: React/Views/RCTView.m [hunk 3]
```diff
@@ -716,6 +716,21 @@ - (NSString *)description
...
+  if (layer.shadowColor != nil && layer.shadowOpacity > 0) {
+    NSShadow *shadow = [NSShadow new];
```
Comment (Saadnajmi) [inline React/Views/RCTView.m:724]: Slight style choice to only have one return, which we prefer: https://microsoft.github.io/objc-guide/MethodsAndImplementations/EarlyReturns.html :

```objective-c

NSShadow *shadow = nil
if (layer.shadowColor != nil && layer.shadowOpacity > 0) {
  shadow = [NSShadow new];
  ...
}
return shadow
```

File: React/Views/RCTView.m [hunk 4]
```diff
@@ -716,6 +716,21 @@ - (NSString *)description
...
+
+    shadow.shadowColor = [[NSColor colorWithCGColor:layer.shadowColor] colorWithAlphaComponent:layer.shadowOpacity];
+    shadow.shadowOffset = layer.shadowOffset;
+    shadow.shadowBlurRadius = layer.shadowRadius;
+
+    return shadow;
+  }
+  return nil;
+}
```
Comment (Saadnajmi) [inline React/Views/RCTView.m:733]: Add an ending `// ]TODO(macOS GH#774)` tag here

**Review by Saadnajmi — APPROVED** [review]

File: React/Views/RCTView.m (hunk 4 above)
Comment (lyzhan7) [reply React/Views/RCTView.m:733]: I think it's already there, after the viewBoundsChanged method

**Pull request merged**